engine/access/state_stream/backend/handler.go

```diff
@@ -259,6 +315,37 @@ func (h *Handler) SubscribeEventsFromLatest(request *executiondata.SubscribeEven
 	return subscription.HandleSubscription(sub, h.handleEventsResponse(stream.Send, request.HeartbeatInterval, request.GetEventEncodingVersion()))
 }
 
+// handleSubscribeExecutionData handles the subscription to execution data and sends it to the client via the provided stream.
+// This function is designed to be used as a callback for execution data updates in a subscription.
+//
+// Parameters:
+// - send: The function responsible for sending execution data in response to the client.
+//
+// Returns a function that can be used as a callback for execution data updates.
+//
+// Expected errors during normal operation:
+//   - codes.Internal - could not convert execution data to entity or could not convert execution data event payloads to JSON.
+func handleSubscribeExecutionData(send sendSubscribeExecutionDataResponseFunc, eventEncodingVersion entities.EventEncodingVersion) func(response *ExecutionDataResponse) error {
+	return func(resp *ExecutionDataResponse) error {
+		execData, err := convert.BlockExecutionDataToMessage(resp.ExecutionData)
+		if err != nil {
+			return status.Errorf(codes.Internal, "could not convert execution data to entity: %v", err)
+		}
+
+		err = convert.BlockExecutionDataEventPayloadsToVersion(execData, eventEncodingVersion)
+		if err != nil {
+			return status.Errorf(codes.Internal, "could not convert execution data event payloads to JSON: %v", err)
+		}
+
+		err = send(&executiondata.SubscribeExecutionDataResponse{
+			BlockHeight:        resp.Height,
+			BlockExecutionData: execData,
```

Comment:
let's also include the block timestamp, it was missing before
https://github.com/onflow/flow/blob/bfe58c88bcb5945c796bfed8eff17b9222ab3fe4/protobuf/flow/executiondata/executiondata.proto#L251-L261